core/src/main/boilerplate/tapir/typelevel/FnComponents.scala.template

```diff
@@ -0,0 +1,29 @@
+package tapir.typelevel;
+
+/*
+ * FN.tupled = TUPLE => RESULT
+ */
+trait FnComponents[-FN, TUPLE, RESULT] {
+  def tupled(f: FN): TUPLE => RESULT
+  def arity: Int
+}
+
+object FnComponents {
+
+  private implicit class HackAddTupledMethodToFunction1[-T1, +R](f: T1 => R) {
```

Comment:
why the hack? :)

Reply:
Because `Function` api is inconsistent and there is no `tupled` method on `Function1`. In order to perform generation we have to start from iteration 1. So it means that the fist generation has to look the same as next generations to certain degree.

Reply:
Ok, I would add a comment here explaining where this comes from :)

Reply:
I prefer code than comments so I tried explaining it using class name :) But I see that I failed.

Reply:
well it says what it does, but doesn't say why ;)